plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -467,7 +470,7 @@ public ConnectorTableHandle getTableHandleForStatisticsCollection(ConnectorSessi
         if (SystemTableHandler.PROPERTIES.matches(tableName)) {
             return getPropertiesSystemTable(session, tableName, SystemTableHandler.PROPERTIES.getSourceTableName(tableName));
         }
-        return Optional.empty();
+        return additionalSystemTableProvider.flatMap(provider -> provider.getSystemTable(session, tableName));
```

Comment:
Instead can you extract system tables we are exposing currently as a `SystemTablesProvider`. And then bind Set<SystemTablesProvider> in connector.

Reply:
Current system tables make use of `SemiTransactionalHiveMetastore` which is created by `io.trino.plugin.hive.HiveMetadataFactory#create()`. I didn't really want to complicate stuff for the same of extension